streaming/base/dataset.py

```diff
@@ -43,12 +43,12 @@ class _ShardState(IntEnum):
 
     Restrictions:
     - The initial state of INVALID must be zero.
-    - State transitions: MISSING -> DOWNLOADING -> PRESENT -> MISSING.
+    - State transitions: REMOTE -> DOWNLOADING -> LOCAL -> REMOTE.
```

Comment:
I liked it.

Reply:
I mean, if a shard is MISSING, but we didn't need it anyway, is it even missed?

If a shard can be PRESENT, should MISSING be "ABSENT"?

Or should shards be RESIDENT or NONRESIDENT?

Or HERE or THERE? Or LOCAL or REMOTE...